test/e2e/framework/clusterrole.go

```diff
@@ -0,0 +1,190 @@
+	ginkgo.By(fmt.Sprintf("Remove sa(%s)", name), func() {
+		clusterClient := GetClusterClient(cluster)
+		gomega.Expect(clusterClient).ShouldNot(gomega.BeNil())
+		err := clusterClient.CoreV1().ServiceAccounts("default").Delete(context.TODO(), name, metav1.DeleteOptions{})
+		gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+	})
+}
+
+// RemoveKarmadalusterSa delete serviceaccount for karmada cluster
+func RemoveKarmadalusterSa(client kubernetes.Interface, name string) {
+	ginkgo.By(fmt.Sprintf("Remove sa(%s)", name), func() {
+		err := client.CoreV1().ServiceAccounts("default").Delete(context.TODO(), name, metav1.DeleteOptions{})
+		gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+	})
+}
+
+// CreateMemberclRo create  clusterrole  with config
+func CreateMemberclusterRole(cluster string, clusterrole *rbacv1.ClusterRole) {
+	ginkgo.By(fmt.Sprintf("Creating member clusterrole(%s)", clusterrole.Name), func() {
+		clusterClient := GetClusterClient(cluster)
+		gomega.Expect(clusterClient).ShouldNot(gomega.BeNil())
+		_, err := clusterClient.RbacV1().ClusterRoles().Create(context.TODO(), clusterrole, metav1.CreateOptions{})
+		gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+	})
+}
+
+//  RemoveMemberclusterRole delete member cluster clusterRole
+func RemoveMemberclusterRole(cluster string, clusterroleName string) {
+	clusterClient := GetClusterClient(cluster)
+	gomega.Expect(clusterClient).ShouldNot(gomega.BeNil())
+	err := clusterClient.RbacV1().ClusterRoles().Delete(context.TODO(), clusterroleName, metav1.DeleteOptions{})
+	if err != nil {
+		fmt.Println(err.Error())
+	}
+}
+
+// CreateMemberclRoBinding create membercluster clusterrolebinding
+func CreateMemberclusterRoleBinding(cluster string, clusterrolebinding *rbacv1.ClusterRoleBinding) {
+	ginkgo.By(fmt.Sprintf("Creating member clusterrolebinding(%s)", clusterrolebinding.Name), func() {
+		clusterClient := GetClusterClient(cluster)
+		gomega.Expect(clusterClient).ShouldNot(gomega.BeNil())
+		_, err := clusterClient.RbacV1().ClusterRoleBindings().Create(context.TODO(), clusterrolebinding, metav1.CreateOptions{})
+		gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+	})
+}
+
+// RemoveMemberclusterRoleBinding delete member cluster clusterRoleBinding
+func RemoveMemberclusterRoleBinding(cluster string, clusterroleBindName string) {
+	clusterClient := GetClusterClient(cluster)
+	gomega.Expect(clusterClient).ShouldNot(gomega.BeNil())
+	err := clusterClient.RbacV1().ClusterRoleBindings().Delete(context.TODO(), clusterroleBindName, metav1.DeleteOptions{})
+	if err != nil {
+		fmt.Println(err.Error())
+	}
+}
+
+// GetSecretInfo get token/url of the karmada cluster
+func GetSecretInfo(client kubernetes.Interface, karmadaClient karmada.Interface, saname string) (token string, url string) {
+	var secretName string
+	sclist := []string{}
+	listSecret, err := client.CoreV1().Secrets("default").List(context.TODO(), metav1.ListOptions{})
```

Comment:
We can get target secret through `ServiceAccount.secrets` field.  And maybe the target secret will take some time to generate, so we need to wait it use `gomega.Eventually`.

Reply:
sorry, i didn't see the comment before  I repush  ,i will fix this  today

Reply:
It's okay.